tests/smoke_tests/verifier.py

```diff
@@ -359,6 +359,215 @@ def _verify(self):
         return True
 
 
+class DataJsonVerifierRateLimited(AgentVerifier):
+    """
+    A verifier that writes 5000 large lines to data.log, then waits until it detects at least one such message in
+    Scalyr, then waits a set time before checking how many lines have been uploaded. The intent of this is to test
+    rate limiting, and as such the agent must be configured with a rate limit that is reflected in
+    `self._rate_limit_bytes_per_second`.
+    """
+
+    def __init__(self, runner, server_address):
+        super(DataJsonVerifierRateLimited, self).__init__(runner, server_address)
+
+        self._data_json_log_path = self._runner.add_log_file(
+            self._runner.agent_logs_dir_path / "data.log"
+        )
+        self._timestamp = datetime.datetime.now().isoformat()
+
+        self._request.add_filter("$serverHost=='{0}'".format(self._agent_host_name))
+        self._request.add_filter(
+            "$logfile=='{0}'".format(
+                self._runner.get_file_path_text(self._data_json_log_path)
+            )
+        )
+        self._request.add_filter("$stream_id=='{0}'".format(self._timestamp))
+
+        self._message = {
+            "count": 0,
+            "stream_id": self._timestamp,
+            "filler": "aaajhghjgfijhgfhhjvcfgujhxfgdtyubn vcgfgyuhbnvcgfytuhvbcftyuhjgftyugftyuyygty7u7y8f8ufgfg8fgf8f"
```

Comment:
To make it slightly easier on the eyes, you could also do ``"aaajhghjgfijhgfhhjvcfgujhxfgdtyubn vcgfgyuhbnvcgfytuhvbcftyuhjgftyugftyuyygty7u7y8f8ufgfg8fgf8f gfyfgufyu" * 60`` :)